hypothesis-python/src/hypothesis/strategies/_internal/types.py

```diff
@@ -594,15 +595,16 @@ def _networks(bits):
     _global_type_lookup[os._Environ] = st.just(os.environ)
 
 
-try:  # pragma: no cover
-    import numpy as np
-
-    from hypothesis.extra.numpy import array_dtypes, array_shapes, arrays, scalar_dtypes
-
-    _global_type_lookup[np.dtype] = array_dtypes()
-    _global_type_lookup[np.ndarray] = arrays(scalar_dtypes(), array_shapes(max_dims=2))
-except ImportError:
-    pass
+# These extras define a callable that either resolves to a strategy for this
+# narrowly extra-specific type, or returns None to proceed with normal type
+# resolution. The callable will only be called if the module is installed. To
+# avoid the performance hit of importing anything here, we defer it until the
+# method is actually called.
+_global_extra_lookup: typing.Dict[str, typing.Callable[[type], st.SearchStrategy]] = {
```

Comment:
```suggestion
_global_extra_lookup: typing.Dict[str, typing.Callable[[type], Optional[st.SearchStrategy]]] = {
```
adding `from typing import ...` at the top would probably be good for readability here.

Reply:
I usually leave `typing.` in, but that's mostly habit. But: do you mean changing all the (many) references to `typing.` in that file, or just `Optional/Dict/Callable`? If the latter IMO it's better to be consistent within the file even if it's a bit more verbose.

Reply:
Ugh, yeah, in this particular file we should probably leave them all as `typing.` - that's distinguishing between `typing`, `typing_extensions`, and possibly `collections.abc` in places 😅